google/cloud/pubsub_v1/subscriber/_protocol/dispatcher.py

```diff
@@ -333,29 +333,35 @@ def modify_ack_deadline(
         # to avoid the server-side max request size limit.
         items_gen = iter(items)
         ack_ids_gen = (item.ack_id for item in items)
-        deadline_seconds_gen = None
-        if default_deadline is None:
-            deadline_seconds_gen = (item.seconds for item in items)
         total_chunks = int(math.ceil(len(items) / _ACK_IDS_BATCH_SIZE))
 
         for _ in range(total_chunks):
             ack_reqs_dict = {
                 req.ack_id: req
                 for req in itertools.islice(items_gen, _ACK_IDS_BATCH_SIZE)
             }
-            # no further work needs to be done for `requests_to_retry`
-            _, requests_to_retry = self._manager.send_unary_modack(
-                modify_deadline_ack_ids=itertools.islice(
-                    ack_ids_gen, _ACK_IDS_BATCH_SIZE
-                ),
-                modify_deadline_seconds=itertools.islice(
-                    deadline_seconds_gen, _ACK_IDS_BATCH_SIZE
+            requests_to_retry: List[requests.ModAckRequest]
+            if default_deadline is None:
+                # no further work needs to be done for `requests_to_retry`
+                _, requests_to_retry = self._manager.send_unary_modack(
+                    modify_deadline_ack_ids=itertools.islice(
+                        ack_ids_gen, _ACK_IDS_BATCH_SIZE
+                    ),
+                    modify_deadline_seconds=itertools.islice(
+                        (item.seconds for item in items), _ACK_IDS_BATCH_SIZE
```

Comment:
This creates an inline generator for every chunk. We want to re-use a generator across chunks so the generator keeps track of where it left off. That's why deadline_seconds_gen is defined outside the for loop.

Reply:
Please check we didn't do this elsewhere.

Reply:
Fixed, and we don't appear to do this elsewhere.